File: TraningAPI/Presenter/RootPresenter/RootPresenter.m
//
//  RootPresenter.m
//  TraningAPI
//
//  Created by 高橋慧 on 2021/06/21.
//  Copyright © 2021 k-takahashi. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "RootPresenter.h"
@implementation RootPresenter
- (instancetype)initWithViewControllerDelegate:(id<RootPresenterOutput>)delegate menuView:(MenuViewController *)menu prefecturesListView:(PrefecturesListViewController *)content{
    self = [self init];
    if (self) {
        self.delegate = delegate;
        menu.delegate = self;
        content.delegate = self;
    }
    return self;
}
/// 選択したメニューのViewController表示
- (void)setMenuID:(int)listIndex {
    [self.delegate performChild:listIndex];
}

/// RootViewControllerの子ビューよりRootViewControllerのメニュースライドボタンの表示・非表示制御
- (void)viewWillAppear {
    [self.delegate toggleIsHidden];
}
@end

File: TraningAPI/Presenter/RootPresenter.swift
//
//  RootPresenter.swift
//  TraningAPI
//
//  Created by 高橋慧 on 2021/05/07.
//  Copyright © 2021 k-takahashi. All rights reserved.
//

import Foundation

protocol RootPresenterOutput:class{
    func performChild(to: Board)
    func toggleIsHidden()
}

protocol RootContainerDelegate:class{
    func setMenuID(to: Board)
    func viewWillAppear()
}

class RootPresenter{
    
    private weak var delegate: RootPresenterOutput?
    init(delegate: RootPresenterOutput, menu: MenuViewController, content: PrefecturesListViewController){
        self.delegate = delegate
        menu.delegate = self
        content.delegate = self
    }
    deinit {
        print("rootpresenter", #function)
    }
}

extension RootPresenter: RootContainerDelegate{
    ///MenuViewControllerのメニューで選択されたBoardを渡す
    func setMenuID(to: Board){
        self.delegate?.performChild(to: to)
    }
    func viewWillAppear(){
        self.delegate?.toggleIsHidden()
    }
}
